client/core/errors.go

```diff
@@ -49,34 +49,56 @@ const (
 	activeOrdersErr
 )
 
-// Error is an error message and an error code.
+// Error is an error message, an error code and a wrapped error.
 type Error struct {
-	s    string
-	code int
+	desc    string
+	code    int
+	wrapped error
 }
 
 // Error returns the error string. Satisfies the error interface.
 func (e *Error) Error() string {
-	return e.s
+	return e.desc
 }
 
+// Code returns the error code.
 func (e *Error) Code() *int {
 	return &e.code
 }
 
+// Unwrap returns the underlying wrapped error.
+func (e *Error) Unwrap() error {
+	return e.wrapped
+}
+
 // newError is a constructor for a new Error.
 func newError(code int, s string, a ...interface{}) error {
-	return &Error{
-		s:    fmt.Sprintf(s, a...),
-		code: code,
+	var err *Error
+	for _, v := range a {
+		e, ok := v.(error)
+		if ok {
+			err = &Error{
+				desc:    fmt.Sprintf(s, a...),
+				code:    code,
+				wrapped: e,
+			}
+			return err
+		}
+	}
+	err = &Error{
+		desc:    fmt.Sprintf(s, a...),
+		code:    code,
+		wrapped: fmt.Errorf(s, a...),
```

Comment:
I don't see why `Error` cannot be this:

```go
type Error struct {
	code int
	err  error
}
```

With `newError` as this:

```go
func newError(code int, s string, a ...interface{}) error {
	return &Error{
		code: code,
		err:  fmt.Errorf(s, a...), // s may contain a %w verb to wrap an error
	}
}
```

and `codedError` as this:

```go
func codedError(code int, err error) error {
	return &Error{
		code: code,
		err:  err,
	}
}
```

and these methods:

```go
func (e *Error) Error() string {
	return e.err.Error()
}
func (e *Error) Unwrap() error {
	return e.err
}
func (e *Error) Code() *int {
	return &e.code
}
```

Reply:
Thanks for the sugestions, all refactoring addressed in latest commit [43ab36c](https://github.com/decred/dcrdex/pull/1511/commits/43ab36c4e82ed1baf1fa700dca613142664b58be)